### PR TITLE
Cip18

### PR DESCRIPTION
**Pull request by chiguireitor**
Up to date status of CIP18, needs bounty address before mergin to master.

### CODE REVIEW

**Comment by droplister** [comment]
**tl;dr**
I ACK this CIP and will [help fund it](https://xchain.io/tx/6c7039f5e8f826cdd82511acab24f8fc224e26a54640592f502c283bcc2032de).

----

But to go into it deeper...

I see that Christian Moss has became the CIP editor yesterday. Awesome! And I see that you're proposing a CIP to the level of it having a number. Cool. I hope to see more CIP proposals receive numbers soon!

But I do not consider this type of change to be something that requires the CIP process be applied.

The current hardcoded limits have worked for nearly five years and "pagination" is available by using the offset parameter.

There are also robust indexing options available through the get_blocks endpoint.

That's my take.

----

To expand on that, I will defend my view by explaining that, in my opinion, if there is no consensus change then that change doesn't rise to the level of a CIP.

The CIP you have created says as much:
> As there's no consensus change in this modification, there's no need for unit testing, however testing on real servers should be done to attest the implementation's fit for the task before accepting the code in master.

I know from testing that, even given limits on certain queries, it is possible to effectively DDOS a federated node through resource-intensive RPC calls. So, in my view, the "default security practice", should be to change your federated node's RPC credentials during the installation phase. 

There is only one known public federated node and that's [public.coindaddy.io](https://coindaddy.io/public-development-servers). So, I wish to note that, it should not be the opinion of the Counterparty community that the "default security practice" for nodes is to allow anonymous public access.

PSA aside, what you are suggesting appears to be a refactoring of the code, which is admirable, but I think those kinds of changes should just be merged during, and in between point releases on the develop branch, rather than become CIPs in and of themselves.

Personally, the way I would implement this change is by updating hardcoded query limits to be configuration values instead that default to the previously hardcoded values. Such a change would be fully backward compatible and give node operators more control. You seem to have designed it as such.

So, given that I DO think that it's a good idea to make federated nodes more configurable... I've had a PR open for about four months to make Counterwallet's code match the published documentation (https://github.com/CounterpartyXCP/counterwallet/issues/870) and there have been other pull requests that I have opened, but had no comments on... And that I'm glad to see ANY CIPs move forward. I am donating the full 75 XCP bounty.

I will help expedite literally any change to the code repositories at this point.

**Comment by droplister** [comment]
> > I am donating the full 75 XCP bounty.
> 
> The 75XCP bounty was already fully funded from the CP dev fund.
> https://xchain.io/tx/1355089

Okie, I guess just send me back the 75 XCP.

**Comment by jdogresorg** [comment]
The full bounty has already been paid out.

https://xchain.io/tx/53746b728133a9b7114553c8026cce75b09f3d582aa7600d167c0b934dbe5ad8

In the past any funds donated to the address assigned to a CIP were automatically sent on to the CIP developer.

**Comment by droplister** [comment]
> The full bounty has already been paid out.
> 
> https://xchain.io/tx/53746b728133a9b7114553c8026cce75b09f3d582aa7600d167c0b934dbe5ad8
> 
> In the past any extra funds donated to the address assigned to a CIP were automatically sent on to the CIP developer.

Oh, bummer. I wish you said before. I was just gonna move it into the Bitcorn Foundation drive thing. All good I guess.
